src/main/java/com/gildedrose/DecreaseBy2.java

```diff
@@ -0,0 +1,7 @@
+package com.gildedrose;
+
+public class DecreaseBy2 implements Updatable {
```

Comment:
This is along the same lines as what was mentioned by @mrfishball and @riccjohn in this thread https://github.com/jramnani/GildedRoseJavaLLC/pull/16/files#r921567623

I'm wondering if at the item level (e.g. "Conjured", "AgedBrie", etc.) you're reaching for inheritance where composition might be more appropriate. Similarly here–I'm wondering if composition, and even the classes, are necessary or if this arithmetic could just live in the implementations of `updateQuality` and `updateSellIn` in each of the respective `Item` classes.

Phrased another way–I'm not sure these classes are buying us much:

`src/main/java/com/gildedrose/DecreaseToZero.java`
`src/main/java/com/gildedrose/DefaultDecrease.java`
`src/main/java/com/gildedrose/DefaultIncrease.java`
`src/main/java/com/gildedrose/IncreaseBy2.java`
`src/main/java/com/gildedrose/IncreaseBy3.java`

Reply:
Yup, I agree. I left a similar comment below, but the idea is the same. I also agree with Eli's idea of just keeping that logic within the respective Item classes for now.

Reply:
+1

Reply:
> I'm wondering if at the item level (e.g. "Conjured", "AgedBrie", etc.) you're reaching for inheritance where composition might be more appropriate.
I agree here. Composition would have been a better call here, idk why my mind was pushing me toward inheritance so much with this abstraction, if we can even call it that.

> if this arithmetic could just live in the implementations of updateQuality and updateSellIn in each of the respective Item classes.
Ok, now I am starting to understand the idea of too much abstraction here. I guess, yes, this would be a good option. It would save us time to get the feature implemented without overoptimizing. Its like if we stopped at like 60% of what we were doing. 

I guess my question to the group here is what is the tradeoff to the approach of not abstracting to the point where we got to? I still don't see much harm in extracting those behaviors to isolate them as an `Update` abstraction. It allows for maximum flexibility with the one behavior whos implementation details are so far the most volatile in the program based on what we have in front of us. It did not really take much longer for us to get to that point. Yes, the classes we created are quite shallow and as the program expands there is a potential to a future developer getting lost in the details but I feel like with proper package setup it would be easy to locate the right class for what you are trying to do.

Reply:
@ikarabulut One of the downsides of this approach is that you'll need a class for every variant, for example, what happens if you need to decrease by 5? or Increase to 20? The other downside is calling `DecreaseBy*` adds a layer of indirection to something that can just be `sellIn - 5` of `quality - 5` without having a clear benefit of either improving reusability or clarity of the code.

Reply:
+1 to @mrfishball. I think this leads to confusing code for other devs to read and it'll become difficult to maintain over time. Even you both as the authors of the code will have a tough time reading it after you haven't touched that part of the codebase for a few weeks or months. If our company merges with another company and suddenly we need to sell 50 new items, this pattern would be really cumbersome.

You mentioned not knowing what's coming next. I 100% agree with that, but since I don't know what types of items are coming next, I'd only do the refactoring that's totally necessary, and wait for patterns to emerge later on as you get new items in the store.

Reply:
The removal of the inheritance and `Updateable` classes in each item is addressed from [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/16/commits/787edac2b69be3565984ac07747d058edd313cba) to [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/16/commits/b4a7a025d8519c2f1355aff41986085493192ee6) (small commits so the whole solution spanned across multiple)